src/AsynchResponders.js

```diff
@@ -1,34 +1,35 @@
-// @flow weak
+// @flow
 class AsynchResponders {
     _responders: Object[];
     constructor() {
         this._responders = [];
     }
 
     // PUBLIC API
-    getLength() {
+    getLength(): number {
         return this._responders.length;
     }
 
-    isBusy() {
+    isBusy(): bool {
         return this.getLength() > 0;
     }
 
-    hasResponder(responder) {
+    hasResponder(responder: any): boolean {
         return this._responders.indexOf(responder) !== -1;
     }
 
-    addResponder(responder) {
+    addResponder(responder: any) {
         this._responders.push(responder);
     }
 
-    addResponders(additionalRespondersArray) {
-        if (additionalRespondersArray && additionalRespondersArray instanceof Array && additionalRespondersArray.length) {
-            this._responders = this._responders.concat(additionalRespondersArray);
+    addResponders(additionalRespondersArray: mixed): void {
+        if (Array.isArray(additionalRespondersArray)) {
+            const arr = additionalRespondersArray.filter((responder: mixed): boolean => typeof responder === 'object' && responder != null);
```

Comment:
As of flow v0.31.0, `Array.isArray()` refines arrays to `Array<mixed>` instead of `Array<any>`, so in order to type-check correctly, you must refine each element of the array.